algosdk/future/transaction.py

```diff
@@ -434,47 +446,196 @@ def dictify(self):
 
         return od
 
+    def __eq__(self, other):
+        if not isinstance(other, (
+                KeyregTxn,
+                transaction.KeyregTxn)):
+            return False
+        return (super(KeyregTxn, self).__eq__(other) and
+                self.votepk == other.votepk and
+                self.selkey == other.selkey and
+                self.votefst == other.votefst and
+                self.votelst == other.votelst and
+                self.votekd == other.votekd and
+                self.nonpart == other.nonpart)
+
+
+class KeyregOnlineTxn(KeyregTxn):
+    """
+    Represents an online key registration transaction.
+    nonpart is implicitly False for this transaction.
+
+    Args:
+        sender (str): address of sender
+        sp (SuggestedParams): suggested params from algod
+        votekey (str): participation public key in base64
+        selkey (str): VRF public key in base64
+        votefst (int): first round to vote
+        votelst (int): last round to vote
+        votekd (int): vote key dilution
+        note (bytes, optional): arbitrary optional bytes
+        lease (byte[32], optional): specifies a lease, and no other transaction
+            with the same sender and lease can be confirmed in this
+            transaction's valid rounds
+        rekey_to (str, optional): additionally rekey the sender to this address
+
+    Attributes:
+        sender (str)
+        fee (int)
+        first_valid_round (int)
+        last_valid_round (int)
+        note (bytes)
+        genesis_id (str)
+        genesis_hash (str)
+        group(bytes)
+        votepk (str)
+        selkey (str)
+        votefst (int)
+        votelst (int)
+        votekd (int)
+        type (str)
+        lease (byte[32])
+        rekey_to (str)
+    """
+
+    def __init__(self, sender, sp, votekey, selkey, votefst,
+                 votelst, votekd, note=None,
+                 lease=None, rekey_to=None):
+        KeyregTxn.__init__(self, sender, sp, votekey, selkey, votefst, votelst,
+                           votekd, note, lease, rekey_to, nonpart=False)
+        self.votepk = votekey
+        self.selkey = selkey
+        self.votefst = votefst
+        self.votelst = votelst
+        self.votekd = votekd
+        if votekey is None:
```

Comment:
@jasonpaulos I added checks here to ensure that a `KeyregOnlineTxn` cannot be initialized when some of the required values are `None`.